assets/js/blocks/single-product/edit/index.tsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { Placeholder, Button, PanelBody } from '@wordpress/components';
+import { withProduct } from '@woocommerce/block-hocs';
+import BlockErrorBoundary from '@woocommerce/base-components/block-error-boundary';
+import EditProductLink from '@woocommerce/editor-components/edit-product-link';
+import { singleProductBlockPreview } from '@woocommerce/resource-previews';
+import { InspectorControls, useBlockProps } from '@wordpress/block-editor';
+import { ProductResponseItem } from '@woocommerce/types';
+import { ErrorObject } from '@woocommerce/editor-components/error-placeholder';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+import ApiError from './api-error';
+import SharedProductControl from './shared-product-control';
+import EditorBlockControls from './editor-block-controls';
+import LayoutEditor from './layout-editor';
+import { BLOCK_ICON } from '../constants';
+import metadata from '../block.json';
+import { Attributes } from '../types';
+
+interface EditorProps {
+	className: string;
+	attributes: {
+		productId: number;
+		isPreview: boolean;
+	};
+	setAttributes: ( attributes: Attributes ) => void;
+	error: string | ErrorObject;
+	getProduct: () => void;
+	product: ProductResponseItem;
+	isLoading: boolean;
+	clientId: string;
+}
+
+/**
+ * Component to handle edit mode of the "Single Product Block".
+ *
+ * @param {Object}            props               Incoming props for the component.
+ * @param {string}            props.className
+ * @param {Object}            props.attributes    Incoming block attributes.
+ * @param {function(any):any} props.setAttributes Setter for block attributes.
+ * @param {string}            props.error
+ * @param {function(any):any} props.getProduct
+ * @param {Object}            props.product
+ * @param {boolean}           props.isLoading
+ * @param {string}            props.clientId
+ */
+const Editor = ( {
+	className,
+	attributes,
+	setAttributes,
+	error,
+	getProduct,
+	product,
+	isLoading,
+	clientId,
+}: EditorProps ) => {
+	const { productId, isPreview } = attributes;
+	const [ isEditing, setIsEditing ] = useState( ! productId );
+	const blockProps = useBlockProps();
+
+	if ( isPreview ) {
+		return singleProductBlockPreview;
+	}
```

Comment:
Should something render as the block preview in the inserter? It doesn't seem to work for me. :confused: 

![imatge](https://user-images.githubusercontent.com/3616980/228863693-43e27349-aac8-4ea1-af45-184384091136.png)

Reply:
I'm going to create a separate PR to solve that because this part is from the old single product block, as soon as we merge this PR I'll start working on the edge cases and enhancements. I put this behind the experimental flag as you suggested

Reply:
Sounds good. :+1: